Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayerSkia.cpp

```diff
@@ -47,7 +47,27 @@
 
 namespace WebCore {
 
-Ref<Nicosia::Buffer> CoordinatedGraphicsLayer::paintTile(const TiledBackingStore& tiledBackingStore, const IntRect& dirtyRect)
+void CoordinatedGraphicsLayer::paintIntoGraphicsContext(GraphicsContext& context, const IntRect& dirtyRect) const
+{
+    IntRect initialClip(IntPoint::zero(), dirtyRect.size());
+    context.clip(initialClip);
+
+    if (!contentsOpaque()) {
+        context.setCompositeOperation(CompositeOperator::Copy);
+        context.fillRect(initialClip, Color::transparentBlack);
+        context.setCompositeOperation(CompositeOperator::SourceOver);
+    }
+
+    auto scale = effectiveContentsScale();
+    FloatRect clipRect(dirtyRect);
+    clipRect.scale(1 / scale);
+
+    context.translate(-dirtyRect.x(), -dirtyRect.y());
+    context.scale(scale);
+    paintGraphicsLayerContents(context, clipRect);
```

Comment:
You are passing in the non-enclosed version of clipRect, whereas we previously passed the rect after mapToContents(), which calls enclosingIntRect. I guess that behavior change wasn't intentional?

Reply:
Yes, it is intentional, because paintGraphicsLayerContents receives a FloatRect, so there's no point in converting to IntRect, no?

Reply:
The old code used `paintGraphicsLayerContents(context, tiledBackingStore.mapToContents(dirtyRect));`, and `mapToContents()` returned `enclosingIntRect(x / scale, y / scale, w / scale, h / scale)`. Now you're passing `(x / scale, y / scale, w / scale, h / scale)` here without taking the enclosingIntRect.

paintGraphicsLayerContents allows clipping on non-integer boundaries, but we previously didn't make use of that -- therefore it's a subtle change in behavior, and we should be sure that it's correct.

Reply:
Ok, let's not change the behavior in a refactoring patch.